In [9]:
import subprocess
import torch
from torch_geometric.data import Dataset, Data
import networkx as nx
import sys

def read(instance_data):
    for line in instance_data.split("\n"):
        if line.startswith("p "):
            _, mode, n, m = line.split()
            assert mode == "min"
            g = nx.DiGraph(mode=mode)
            n = int(n)
            # if "GRIDGRAPH" in filename:
            #    k=0
            # else:
            k = 1
            nodes = list(range(k, n + k))
            g.add_nodes_from(nodes, demand=0)
        elif line.startswith("n "):
            _, ID, demand = line.split()
            g.nodes[int(ID)]["demand"] = int(demand)
        elif line.startswith("a "):
            _, src, dest, low, cap, cost = line.split()
            g.add_edge(int(src), int(dest), capacity=int(cap), weight=int(cost))
            assert int(low) == 0

    return g

In [13]:
command = "python generate_gridgraph.py 18 30 447 10000 1000 13670324914702", ""
instance_data = subprocess.run(command[0].replace("python", sys.executable), capture_output=True, text=True, shell=True, input=command[1]).stdout
G = read(instance_data)

In [27]:
%load_ext autoreload
%autoreload 2
import torch_loader
loader = torch_loader.MCFDataset("../data/generated_data/")

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


659

In [71]:
loader.get(0).edge_attr.shape[0]

125152

In [ ]:
import time
now = time.time()
nodes = []
edges = []
for i in range(len(loader)):
    graph_i = loader.get(i)
    print(f"{i}", graph_i)
    nodes.append(graph_i.num_nodes)
    edges.append(graph_i.edge_attr.shape[0])

print(time.time() - now)
print(f"Avg. nodes {np.mean(nodes)}")
print(f"Avg. edges {np.mean(edges)}")

0 Data(edge_index=[2, 125152], label=[4], num_nodes=3911, x=[3911, 1], edge_attr=[125152, 2])
1 Data(edge_index=[2, 2984], label=[4], num_nodes=1510, x=[1510, 1], edge_attr=[2984, 2])
2 Data(edge_index=[2, 45359], label=[4], num_nodes=2276, x=[2276, 1], edge_attr=[45359, 2])
3 Data(edge_index=[2, 100777], label=[4], num_nodes=2870, x=[2870, 1], edge_attr=[100777, 2])
4 Data(edge_index=[2, 14700], label=[4], num_nodes=1225, x=[1225, 1], edge_attr=[14700, 2])
5 Data(edge_index=[2, 101035], label=[4], num_nodes=3559, x=[3559, 1], edge_attr=[101035, 2])
6 Data(edge_index=[2, 26388], label=[4], num_nodes=2199, x=[2199, 1], edge_attr=[26388, 2])
7 Data(edge_index=[2, 105145], label=[4], num_nodes=3369, x=[3369, 1], edge_attr=[105145, 2])
8 Data(edge_index=[2, 48182], label=[4], num_nodes=1267, x=[1267, 1], edge_attr=[48182, 2])
9 Data(edge_index=[2, 163248], label=[4], num_nodes=3401, x=[3401, 1], edge_attr=[163248, 2])
10 Data(edge_index=[2, 61875], label=[4], num_nodes=375, x=[375, 1], edg

|Name|Graphs|Classes|Avg. Nodes|Avg. Edges|Node Labels|Edge Labels|Node Attr.|Edge Attr.|Geometry|
|---|---|---|---|---|---|---|---|---|---|
|MCFSmall|659|4|0|0|-|-| +(1)| +(2)| - |
